### ISSUE TITLE
gob: type elliptic.p256Curve has no exported fields

### ISSUE THREAD

**Issue by ghost** [Jeiwan/blockchain_go#85]
part_5 createwallet not working @Jeiwan

**Comment by Anmilover** [comment]
您好，邮件已查收。Anmi、

**Comment by BryceWayne** [comment]
@mithleshmeghwal0 Which code section?

**Comment by ghost** [comment]
@BryceWayne 
https://github.com/Jeiwan/blockchain_go/blob/f35aa3c3cacd9be7b2a6f31d0f550bcaf31c6e53/wallets.go#L87
its failing on this line
Screeshot is from my local, so line number may differ
<img width="735" alt="Screenshot 2565-12-26 at 11 34 34" src="https://user-images.githubusercontent.com/115926756/209500389-699b9074-04aa-4981-a9e7-f2b4f5393881.png">

**Comment by BryceWayne** [comment]
`elliptic.p256Curve` should have no exported fields, right? Because of `p` being lowercase. Inspecting further.

**Comment by BryceWayne** [comment]
Here is why. https://cs.opensource.google/go/go/+/refs/tags/go1.19.4:src/crypto/elliptic/nistec.go;l=35;drc=38cfb3be9d486833456276777155980d1ec0823e

The field is not accessible outside the file.

**Comment by ghost** [comment]
Yup, i inspected until this point, but can we do any work around this ?

**Comment by BryceWayne** [comment]
For context, https://stackoverflow.com/questions/32676898/whats-the-purpose-of-gob-register-method
The problem is saving the wallets map so that we can always reconstruct the data... Just thinking out loud.

**Comment by BryceWayne** [comment]
One way might be to create your own type/interface to access to field...
https://stackoverflow.com/questions/42664837/how-to-access-unexported-struct-fields

**Comment by ghost** [comment]
Yup, i will try to create interface and i will comment or probably create PR for this . Thanks @BryceWayne

**Comment by ghost** [comment]
@BryceWayne  i tried to build a package with elliptic , but it still failed
@Jeiwan

**Comment by LangLiBa** [comment]
> @BryceWayne
> 
> https://github.com/Jeiwan/blockchain_go/blob/f35aa3c3cacd9be7b2a6f31d0f550bcaf31c6e53/wallets.go#L87
> 
> 
> its failing on this line
> Screeshot is from my local, so line number may differ
> <img alt="Screenshot 2565-12-26 at 11 34 34" width="735" src="https://user-images.githubusercontent.com/115926756/209500389-699b9074-04aa-4981-a9e7-f2b4f5393881.png">

Did you fix these,please?

**Comment by Anmilover** [comment]
您好，邮件已查收。Anmi、

**Comment by ghost** [comment]
Seems @LangLiBa this repo is outdated, i created this issue long ago, there is no reply. I fixed this by using 1.18.9 go version. 
I could have pursued to fix this, but my objective was to learn blockchain, not to fix this

**Comment by LangLiBa** [comment]
> Seems @LangLiBa this repo is outdated, i created this issue long ago, there is no reply. I fixed this by using 1.18.9 go version. I could have pursued to fix this, but my objective was to learn blockchain, not to fix this

Ok, thanks a lot.

**Comment by BryceWayne** [comment]
> Seems @LangLiBa this repo is outdated, i created this issue long ago, there is no reply. I fixed this by using 1.18.9 go version. I could have pursued to fix this, but my objective was to learn blockchain, not to fix this

Thanks @mithleshmeghwal0

**Comment by zzylmt** [comment]
I'm getting the same error, how should I fix this problem

**Comment by Anmilover** [comment]
您好，邮件已查收。Anmi、

**Comment by mithRabbit** [comment]
> 您好，邮件已查收。Anmi、

use go version 1.18.9

**Comment by mithRabbit** [comment]
> I'm getting the same error, how should I fix this problem

use go version 1.18.9

**Comment by zzylmt** [comment]
Is there another solution? Or please update the code based on the latest go version？thx.

**Comment by yohlime** [comment]
Here is my [workaround](https://github.com/atsuyaourt/blockchain_go/blob/part5/internal/blockchain/wallet.go#L89-L143) (see lns 89-143). Then remove ln 66 and ln 82 from [wallets.go](https://github.com/Jeiwan/blockchain_go/blob/part_5/wallets.go).

I implemented GobEncoder/GobDecoder interfaces to the `Wallet` struct so `gob.Register(elliptic.P256())` can be removed.

**Comment by Anmilover** [comment]
您好，邮件已查收。Anmi、

**Comment by kiwi100** [comment]
> Here is my [workaround](https://github.com/atsuyaourt/blockchain_go/blob/part5/internal/blockchain/wallet.go#L89-L143) (see lns 89-143). Then remove ln 66 and ln 82 from [wallets.go](https://github.com/Jeiwan/blockchain_go/blob/part_5/wallets.go).
> 
> I implemented GobEncoder/GobDecoder interfaces to the `Wallet` struct so `gob.Register(elliptic.P256())` can be removed.

Thanks, it worked.
